flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -103,10 +107,26 @@
   private static final ListStateDescriptor<SortedMap<Long, List<DataFile>>> STATE_DESCRIPTOR = buildStateDescriptor();
   private transient ListState<SortedMap<Long, List<DataFile>>> checkpointsState;
 
-  IcebergFilesCommitter(TableLoader tableLoader, Configuration hadoopConf, boolean replacePartitions) {
+  IcebergFilesCommitter(TableLoader tableLoader, Configuration hadoopConf, boolean replacePartitions,
+                        org.apache.flink.configuration.Configuration flinkConf) {
     this.tableLoader = tableLoader;
     this.hadoopConf = new SerializableConfiguration(hadoopConf);
     this.replacePartitions = replacePartitions;
+    this.flinkConf = flinkConf;
+    this.flushCommitInterval = flinkConf.getLong(FlinkSink.FLINK_ICEBERG_SINK_FLUSHINTERVAL,
+        FlinkSink.DEFAULT_FLINK_ICEBERG_SINK_FLUSHINTERVAL);
+  }
+
+  @Override
+  public void open() throws Exception {
+    super.open();
+    boolean isCheckpointEnabled = getRuntimeContext().isCheckpointingEnabled();
+    // If we don't enable checkpoint, we will use processingTimeSerice to do commit,
+    if (!isCheckpointEnabled) {
+      ProcessingTimeService processingTimeService = getRuntimeContext().getProcessingTimeService();
+      final long currentTimestamp = processingTimeService.getCurrentProcessingTime();
+      processingTimeService.registerTimer(currentTimestamp + flushCommitInterval, this);
+    }
   }
 
   @Override
```

Comment:
We will invoke the `initializeState` method even if disabled the flink checkpoint ?  If not, then many variables (such as table , maxCommittedCheckpointId)  won't be initialized, that would introduce unexpected bug.

Reply:
Yes initializeState will invoke even checkpoint disabled, and context.isRestored will be false.